iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -186,10 +186,13 @@ inline bool vector<T, Capacity>::emplace(const uint64_t position, Targs&&... arg
     {
         return emplace_back(std::forward<Targs>(args)...);
     }
-
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by template parameter
+    // NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
     emplace_back(std::move(*reinterpret_cast<T*>(m_data[m_size - 1U])));
```

Comment:
@elfenpiff A big downside to using an aligned, underlying byte array together with `reinterpret_cast`, is that the `cxx::vector` can never become `constexpr` when we move to C++17. [This talk](https://www.youtube.com/watch?v=I8QJLGI0GOE) provides some alternative ideas.

However, I suppose we'll stick with the current approach for now. Instead of implementing `uninitialized_array`, we could think about using `std::aligned_storage`.

Reply:
@mossmaurice  AFAIK the problem is the destructor not the reinterpret cast. I'm not quite sure if we can easily circumvent this. One option is of course a specialization for trivially destructible types but that might be quite an amount of work.

Reply:
@elBoberido David Stone says something different in the talk and see [this thread](https://stackoverflow.com/a/59913380/18276160). `reinterpret_cast`s are not possible in `constexpr` methods during compile-time.

Reply:
oh boy, I just love this language :grimacing: